plugin/src/lib.rs

```diff
@@ -724,6 +724,23 @@ fn fold_first_block(block: P<Block>, m: &mut MutatorPlugin) -> P<Block> {
                         .unwrap(),
                 );
             }
+            let (ref key, ref values) = for (key, &value) in interchangeables.iter() {
+                let n = *current_count;
+                let ident = key.to_ident();
+                add_mutations(
+                    cx,
+                    mutations,
+                    current_count,
+                    block.span,
+                    &["insert return default()"],
+                );
+                if mutagen::now(n) {
+                    (value, key)
+                } else {
+                    (key, value)
+                };
```

Comment:
I'm getting here an error

```
if and else have incompatible types
   --> src/lib.rs:737:17
    |
737 | /                 if ::mutagen::now(n) {
738 | |                     (value, key)
739 | |                 } else {
740 | |                     (key, value)
741 | |                 };
    | |_________________^ expected struct `std::vec::Vec`, found reference
```

Reply:
Yes, you need this within a `quote!`(same as with the other mutations). So wrap it in `quote!(cx, /*YOUR CODE HERE*/)` and change the `n` to `$n` (this gets our mutation number into the generated code), and change `key` and `value` to `$key_ident` and `$value_ident` respectively. Now you only need to prepend the `let ($key_ident, $value_ident) = ` before that `if` so the mutation will actually work.